lldb/source/Plugins/LanguageRuntime/Swift/SwiftTask.h

```diff
@@ -0,0 +1,93 @@
+
+#include "lldb/Target/RegisterContext.h"
+#include "lldb/Target/Thread.h"
+#include "lldb/Utility/RegisterValue.h"
+#include "lldb/lldb-forward.h"
+
+namespace lldb_private {
+
+using namespace lldb;
+
+/// Provides a subset of Thread operations for Swift Tasks.
+///
+/// Currently, this supports backtraces of Tasks, and selecting frames in the
+/// backtrace. Async frames make available the variables that are stored in the
+/// Task's "async context" (instead of the stack).
+///
+/// See `Task<Success, Failure>` and `UnsafeCurrentTask`
+class ThreadTask : public Thread {
+public:
+  ThreadTask(tid_t tid, addr_t async_ctx, ExecutionContext &exe_ctx);
+
+  /// Returns a Task specific register context (RegisterContextTask).
+  RegisterContextSP GetRegisterContext() override;
+
+  ~ThreadTask() override { DestroyThread(); }
+
+  // No-op overrides.
+  void RefreshStateAfterStop() override {}
+  lldb::RegisterContextSP
+  CreateRegisterContextForFrame(StackFrame *frame) override {
+    return {};
+  }
+  bool CalculateStopInfo() override { return false; }
+
+private:
+  /// A register context that is the source of `RegisterInfo` data.
+  RegisterContextSP m_concrete_reg_ctx_sp;
+  /// Lazily initialized `RegisterContextTask`.
+  RegisterContextSP m_async_reg_ctx_sp;
+  /// The Task's async context.
+  addr_t m_async_ctx = LLDB_INVALID_ADDRESS;
+  /// The address of the async context's resume function.
+  addr_t m_pc = LLDB_INVALID_ADDRESS;
```

Comment:
The name `pc` here is a bit misleading: it would be natural to think this contains the PC of the top frame of the thread, which is _not_ what this variable contains.
Maybe it would be better to call this `m_continuation_address`?

When we are unwinding, we say "pc" because we are producing the register context of the frame above the current frame. Here, we just have a Thread class with a member `pc`, so I don't think the same interpretation would apply.

Reply:
Actually I am confused now, is this really printing frame 0 correctly?
In the register context class, there is this code:

```
  if (reg_info->kinds[eRegisterKindGeneric] == LLDB_REGNUM_GENERIC_PC) {
     reg_value = m_pc;
     return true;
   }
```

I could be missing something, but this looks like it prints the PC of frame 1 instead of frame 0.

Reply:
maybe `m_resume_address`, since "resume" is the term used in `AsyncContext`.

I'm not sure where the confusion stems from, but yes this prints "frame 0" correctly.